packages/sanity/src/core/form/inputs/CrossDatasetReferenceInput/useReferenceInfo.ts

```diff
@@ -72,14 +71,6 @@ export function useReferenceInfo(
           )
         : of(EMPTY_STATE),
     [docInfo, getReferenceInfo, retry, retryAttempt],
-    INITIAL_LOADING_STATE,
   )
-
-  // workaround for a "bug" with useMemoObservable that doesn't
-  // return the initial value upon resubscription
-  const previousId = usePrevious(doc._id, doc._id)
-  if (doc._id && previousId !== doc._id) {
-    return INITIAL_LOADING_STATE
-  }
```

Comment:
@bjoerge can you verify that this is fixed?

Reply:
Tested & verified